components/track/TrackShareDialog.vue

```diff
@@ -0,0 +1,93 @@
+<script lang="ts" setup>
+// import type { NuxtIconName } from '.nuxt/nuxt-icons';
+
+const { t } = useI18n();
+
+const props = defineProps<{
+  trackId: number;
+}>();
+
+const emit = defineEmits<{
+  close: [];
+}>();
+
+const message = ref<string | null>(null);
+
+// TODO: Make this dynamic
+const trackLink = `http://localhost:9001/track/${props.trackId}`;
+
+const setMessage = (newMessage: string) : Promise<null> => new Promise((resolve) => {
+    message.value = newMessage;
+    setTimeout(() => {
+      message.value = null;
+      resolve(null);
+    }, 2000);
+  });
+
+const shareButtons = {
```

Comment:
I'd define a type here which only allows two types: One with `link` and the other one with `clickFunction`.

Take a look at the file `TrackMenu.vue` for an example. This would make your button down type-safe without optional chaining...